app/src/components/generative/ModelMenu.tsx

```diff
@@ -139,6 +175,136 @@ export function ModelMenu({ value, onChange }: ModelMenuProps) {
   );
 }
 
+type ModelsByProviderMenuProps = {
+  modelsByProvider: Map<string, string[]>;
+  providerInfoMap: Map<
+    string,
```

Comment:
is the use of `string` type deliberate?
```suggestion
  modelsByProvider: Map<GenerativeProviderKey, string[]>;
  providerInfoMap: Map<
    GenerativeProviderKey,
```